### PR TITLE
Eror404

### PR DESCRIPTION
**Pull request by lizzav**

### CODE REVIEW

File: src/services/routeConfig.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import TestsPage from '../pages/testsPage';
 import DisciplinePage from '../pages/disciplinePage';
 import TestInfo from '../components/TestInfo';
 import TestSolution from '../components/TestSolution';
+import ErrorPage from '../pages/errorPage/errorPage';
```
Comment (kaspian122) [inline src/services/routeConfig.js]: import ErrorPage from '../pages/ErrorPage'

Comment (VechniiNewbie) [reply src/services/routeConfig.js:11]: pages should be in pages folder

File: src/constants/routerPaths.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ const RouterPaths = {
   tests: '/tests',
   loginPage: '/login',
   registration: '/registration',
-  errorPage: '/error',
```
Comment (tchernykhds) [inline src/constants/routerPaths.js:10]: If we caught an error, the app should redirect us to the error page with '/404' endpoint or something

Comment (tchernykhds) [reply src/constants/routerPaths.js:10]: @kaspian122 what ya think

File: src/pages/errorPage/errorPage.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    history.push(`/disciplines`);
+  }, [history]);
+  return (
+    <div className="error-page">
+      <div className="error-page__text">
+        <span className="error-page__text_title">404</span>
+        <span>Страница не найдена</span>
+      </div>
+      <div className="error-page__button" onClick={handleDisciplinesClick}>
+        Вернуться на главную(?){' '}
```
Comment (tchernykhds) [inline src/pages/errorPage/errorPage.jsx]: why is the space (`{' '}`) here?

File: src/pages/errorPage/errorPage.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import { useHistory } from 'react-router-dom';
+import { useDidMount } from '../../utils/hooks';
+import './errorPage.scss';
+function ErrorPage({ setTitle = () => {} }) {
+  const history = useHistory();
+  useDidMount(() => {
+    setTitle('');
+  });
+  const handleDisciplinesClick = useCallback(() => {
+    history.push(`/disciplines`);
```
Comment (tchernykhds) [inline src/pages/errorPage/errorPage.jsx]: I think the constant should be here `RouterPaths.disciplines`

File: src/pages/errorPage/errorPage.scss [hunk 5]
```diff
@@ -0,0 +1,33 @@
+@import 'src/scss/vars';
+
+.error-page {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin-top: 15%;
+  height: 273px;
+  justify-content: space-between;
+  &__text {
```
Comment (tchernykhds) [inline src/pages/errorPage/errorPage.scss]: space before nested block

File: src/pages/errorPage/errorPage.scss [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    height: 273px;
+    :last-child {
+      margin-top: 25px;
+    }
+    &_title {
```
Comment (tchernykhds) [inline src/pages/errorPage/errorPage.scss]: space between sibling blocks

File: src/services/routeConfig.js (hunk 1 above)
Comment (tchernykhds) [reply src/services/routeConfig.js]: maybe a page should be in the pages folder?

File: src/components/App.jsx [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import './App.scss';
-import React, { useState } from 'react';
+import React, { useCallback, useState } from 'react';
```
Comment (tchernykhds) [inline src/components/App.jsx]: excess import and for what?

File: src/pages/errorPage/errorPage.jsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import React, { useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useDidMount } from '../../utils/hooks';
+import './errorPage.scss';
```
Comment (tchernykhds) [inline src/pages/errorPage/errorPage.jsx]: import code block should be separated from the other part of the file by a space

**Review by tchernykhds — CHANGES_REQUESTED** [review]

**Pull request merged**